### ISSUE TITLE
Undeleteable files and directories

### ISSUE THREAD

**Issue by GoogleCodeExporter** [XianglinLi/opendedup#10]
```
SDFS 0.8.12, Debian Lenny++.

Some files and directories are undeleteable.  A directory:

  hypatia:/backup/current# rm -r
./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps
  rm: cannot remove directory
`./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps': Function not
implemented
  hypatia:/backup/current# ls -ld
./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps
  drwxr-xr-x 1 4550 root 0 Nov 20  2008
./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps
  hypatia:/backup/current# ls -la
./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps
  total 0
  hypatia:/backup/current# rmdir
./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps
  rmdir: failed to remove
`./fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/minidumps': Function not
implemented

Files:
  hypatia:/backup/current# rm -r fklfp01-nor
  rm: cannot remove `fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bilder/Kan
slettes.txt': Not a directory
  rm: cannot remove `fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/key3.db':
Not a directory
  rm: cannot remove
`fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bokmerke/places.sqlite': Not a directory
  hypatia:/backup/current# ls -l
fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bilder/Kan\ slettes.txt
  -rw------- 1 600 root 0 Aug 11  2009
fklfp01-nor/HOMES/NOR/ELEV/mabaxxxx/Bilder/Kan slettes.txt
  hypatia:/backup/current# ls -ld
fklfp01-nor/HOMES/NOR/ELEV/mabaxxx/Bokmerke/key3.db
  -rw------- 1 4550 root 16384 May  4  2009
fklfp01-nor/HOMES/NOR/ELEV/mabaxxx/Bokmerke/key3.db
  hypatia:/backup/current# ls -ld
fklfp01-nor/HOMES/NOR/ELEV/mabaxxx/Bokmerke/places.sqlite
  -rw------- 1 4550 root 262144 May  4  2009
fklfp01-nor/HOMES/NOR/ELEV/mabaxxx/Bokmerke/places.sqlite

A reproducible case:
  hypatia:/backup/current# mkdir dir1 dir2
  hypatia:/backup/current# echo foo > dir1/bar
  hypatia:/backup/current# echo bar > dir2/bar
  hypatia:/backup/current# rm -r dir?
  rm: cannot remove `dir1/bar': Not a directory
  rm: cannot remove `dir2/bar': Not a directory
  hypatia:/backup/current# mkdir dir3 dir4
  hypatia:/backup/current# echo foo > dir3/gazonk
  hypatia:/backup/current# echo foo > dir4/zzz
  hypatia:/backup/current# rm -r dir3 dir4
  rm: cannot remove `dir3/gazonk': Not a directory
  rm: cannot remove `dir4/zzz': Not a directory
  hypatia:/backup/current# rm dir?/*
  hypatia:/backup/current# rmdir dir?
  rmdir: failed to remove `dir1': Function not implemented
  rmdir: failed to remove `dir2': Function not implemented
  rmdir: failed to remove `dir3': Function not implemented
  rmdir: failed to remove `dir4': Function not implemented

Two issues:
o rm -r think files are directories.  Same thing happens to mv(1).
o Directories can't be deleted, even empty ones.
```

Original issue reported on code.google.com by `stur...@gmail.com` on 31 Mar 2010 at 1:04

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 0.8.13 line. 

http://opendedup.googlecode.com/files/sdfs-0.8.13-unstable.tar.gz
```

Original comment by `sam.silv...@gmail.com` on 31 Mar 2010 at 8:52
- Changed state: **Verified**

**Comment by GoogleCodeExporter** [comment]
```
0.8.13
```

Original comment by `sam.silv...@gmail.com` on 1 Apr 2010 at 6:11
- Changed state: **Fixed**
